earth_enterprise/src/common/khxml/khxml.cpp

```diff
@@ -161,18 +220,38 @@ class UsingXMLGuard
 };
 
 static khMutexBase xmlLibLock = KH_MUTEX_BASE_INITIALIZER;
+static khMutexBase checkTermLock = KH_MUTEX_BASE_INITIALIZER;
+static khMutexBase cacheLock = KH_MUTEX_BASE_INITIALIZER;
+
+// only terminate when certain conditions are met
+void  readyForTerm()
+{
+  khLockGuard guard(checkTermLock);
+  static XMLSSize_t threashold = static_cast<XMLSSize_t>
+         (maxDOMHeapAllocSize * percent); 
+  if (cacheCapacity >= threashold)
```

Comment:
Thinking out loud: it might make things simpler to remove the threshold code and simply terminate the cache every time there's an opportunity to do so. That would remove some of the code and the need to stat the file. On the other hand, it's nice to have a way to tune the terminate behavior.